app/code/Magento/InventoryCatalogSearch/etc/di.xml

```diff
@@ -9,4 +9,12 @@
     <type name="Magento\CatalogSearch\Model\Search\FilterMapper\StockStatusFilter">
         <plugin name="adapt_stock_status_filter" type="Magento\InventoryCatalogSearch\Plugin\Search\FilterMapper\AdaptStockStatusFilterPlugin"/>
     </type>
+    <type name="Magento\CatalogSearch\Model\Adapter\Mysql\Aggregation\DataProvider\SelectBuilderForAttribute\JoinStockConditionToSelect">
+        <plugin name="adapt_data_provider_join_stock_condition_to_select"
```

Comment:
Pls change keys according to plugin name

Reply:
Fixed.